ui/src/store/catalog.test.ts

```diff
@@ -15,62 +20,57 @@ describe('Store Object', () => {
   });
 
   it('creates a catalog store', (done) => {
-    const store = CatalogStore.create({});
-
-    const item = Catalog.create({
-      id: 1,
-      name: 'tekton',
-      type: 'community'
-    });
-
-    store.add(item);
-
-    expect(getSnapshot(store.items)).toMatchSnapshot();
-
-    done();
+    const store = CatalogStore.create({}, { api });
+    expect(store.isLoading).toBe(true);
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.items.size).toBe(2);
+        expect(store.isLoading).toBe(false);
+
+        expect(getSnapshot(store)).toMatchSnapshot();
+
+        done();
+      }
+    );
   });
 
   it('should toggle a selected catalog', (done) => {
-    const store = CatalogStore.create({});
-
-    const item = Catalog.create({
-      id: 1,
-      name: 'tekton',
-      type: 'community'
-    });
-
-    store.add(item);
-
-    const catalogs = store.items.get('1');
-    assert(catalogs);
-    catalogs.toggle();
-
-    expect(store.selected.size).toBe(1);
-    expect(catalogs.selected).toBe(true);
-
-    done();
+    const store = CatalogStore.create({}, { api });
+    expect(store.isLoading).toBe(true);
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.isLoading).toBe(false);
+        const catalogs = store.items.get('1');
+        assert(catalogs);
+        catalogs.toggle();
+
+        expect(store.selected.size).toBe(1);
+        expect(catalogs.selected).toBe(true);
+
+        done();
+      }
+    );
   });
 
   it('should clear all the selected catalog', (done) => {
-    const store = CatalogStore.create({});
-
-    const item = Catalog.create({
-      id: 1,
-      name: 'tekton',
-      type: 'community'
-    });
-
-    store.add(item);
-
-    const catalogs = store.items.get('1');
-    assert(catalogs);
-    catalogs.toggle();
-
-    store.clearSelected();
-
-    expect(catalogs.selected).toBe(false);
-
-    done();
+    const store = CatalogStore.create({}, { api });
+    when(
+      () => !store.isLoading,
+      () => {
+        expect(store.isLoading).toBe(false);
+
+        const catalogs = store.items.get('1');
+        assert(catalogs);
+        catalogs.toggle();
+
+        store.clearSelected();
+
+        expect(catalogs.selected).toBe(false);
+        done();
+      }
+    );
   });
 
   it('can get an icon for catalog', (done) => {
```

Comment:
Why are we not injecting `api` while creating the store for other tests ?

Reply:
not required, but needs to modify test